File: AdisyonUygulamaJavaXMLsiz/app/src/main/java/com/example/adisyonuygulamajava/services/Services.java
package com.example.adisyonuygulamajava.services;

import com.example.adisyonuygulamajava.model.Kategori;
import com.example.adisyonuygulamajava.model.Masa;
import com.example.adisyonuygulamajava.model.MasaUrun;
import com.example.adisyonuygulamajava.model.Urun;
import com.example.adisyonuygulamajava.services.response.KategoriSilResponse;
import com.example.adisyonuygulamajava.services.response.UrunSilResponse;

import java.util.List;

import okhttp3.ResponseBody;

public interface Services {

    void urunEkle(String urunAd, float fiyat, int kategoriId, int adet, String base64);

    ResponseBody masaSil(int masaId);

    ResponseBody masaEkle();

    List<Masa> masalariGetir();

    List<Urun> urunleriGetir();

    List<MasaUrun> masaUrunleriniGetir(int masaId);

    ResponseBody urunEkle(int masaId, int urunId, int adet);

    ResponseBody masaOdemeYap(int masaId);

    float masaToplamFiyat(int masaId);

    ResponseBody urunCikar(int masaId, int urunId);

    List<Kategori> kategorileriGetir();

    Masa masaGetir(int masaId);

    ResponseBody masaBirlestir(int anaMasaId, int birlestirilecekMasaId);

    ResponseBody kategoriEkle(String ad);

    KategoriSilResponse kategoriSil(int id);

    UrunSilResponse urunSil(String urunAd);
}

File: AdisyonUygulamaJetpack/app/src/main/java/com/example/adisyonuygulamajetpack/services/Services.kt
package com.example.qrkodolusturucu.services

import com.example.adisyonuygulamakotlin.model.Kategori
import com.example.adisyonuygulamakotlin.model.Masa
import com.example.adisyonuygulamakotlin.model.MasaUrun
import com.example.adisyonuygulamakotlin.model.Urun
import com.example.adisyonuygulamakotlin.services.response.KategoriSilResponse
import com.example.adisyonuygulamakotlin.services.response.UrunSilResponse
import okhttp3.ResponseBody


interface Services {
    suspend fun urunEkle(urunAd: String, fiyat: Float, kategoriId: Int, adet: Int, base64: String)
    suspend fun masaSil(masaId: Int):ResponseBody
    suspend fun masaEkle():ResponseBody
    suspend fun masalariGetir(): List<Masa>
    suspend fun urunleriGetir(): List<Urun>
    suspend fun masaUrunleriniGetir(masaId: Int): List<MasaUrun>
    suspend fun urunEkle(masaId: Int, urunId: Int, adet: Int): ResponseBody
    suspend fun masaOdemeYap(masaId: Int): ResponseBody
    suspend fun masaToplamFiyat(masaId: Int): Float
    suspend fun urunCikar(masaId: Int, urunId: Int): ResponseBody
    suspend fun kategorileriGetir(): ArrayList<Kategori>
    suspend fun masaGetir(masaId: Int): Masa
    suspend fun masaBirlestir(anaMasaId: Int,birlestirilecekMasaId: Int): ResponseBody
    suspend fun kategoriEkle(ad: String): ResponseBody
    suspend fun kategoriSil(id: Int): KategoriSilResponse
    suspend fun urunSil(urunAd: String): UrunSilResponse
}